src/scales/scale.radialLinear.js

```diff
@@ -47,10 +47,300 @@ module.exports = function(Chart) {
+		 * on each side, removing that from the size, halving it and adding the left x protrusion width.
+		 *
+		 * This will mean we have a shape fitted to the canvas, as large as it can be with the labels
+		 * and position it in the most space efficient manner
+		 *
+		 * https://dl.dropboxusercontent.com/u/34601363/yeahscience.gif
+		 */
+
+		var plFont = getPointLabelFontOptions(scale);
+
+		// Get maximum radius of the polygon. Either half the height (minus the text width) or half the width.
+		// Use this to calculate the offset + change. - Make sure L/R protrusion is at least 0 to stop issues with centre points
+		var largestPossibleRadius = Math.min(scale.height / 2, scale.width / 2),
+			i,
+			textSize,
+			pointPosition,
+			furthestLeft = scale.width,
+			furthestRight = 0,
+			furthestAbove = scale.height,
+			furthestBelow = 0,
+			furthestLeftAngle,
+			furthestRightAngle,
+			furthestAboveAngle,
+			furthestBelowAngle,
+			newLeft,
+			newRight,
+			newTop,
+			newBottom;
+		scale.ctx.font = plFont.font;
+		scale._pointLabelSizes = [];
+
+		function determineXLimits(angle) {
+			if (angle === 0 || angle === 180) {
+				// center X
+				newLeft = pointPosition.x - (textSize.w / 2);
+				newRight = pointPosition.x + (textSize.w / 2);
+			} else if (angle < 180) {
+				// text on right side
+				newLeft = pointPosition.x;
+				newRight = pointPosition.x + textSize.w + 5;
+			} else {
+				// text on left side
+				newLeft = pointPosition.x - textSize.w - 5;
+				newRight = pointPosition.x;
+			}
+		}
+
+		function determineYLimits(angle) {
+			if (angle === 90 || angle === 270) {
+				// center Y
+				newTop = pointPosition.y - (textSize.h / 2);
+				newBottom = pointPosition.y + (textSize.h / 2);
+			} else if (angle > 270 || angle < 90) {
+				// text in top half
+				newTop = pointPosition.y - textSize.h - 5;
+				newBottom = pointPosition.y;
+			} else {
+				// text in bottom half
+				newTop = pointPosition.y;
+				newBottom = pointPosition.y + textSize.h + 5;
+			}
+		}
+
+		function updateLimits(angleRadians) {
+			if (newLeft < furthestLeft) {
+				furthestLeft = newLeft;
+				furthestLeftAngle = angleRadians;
+			}
+
+			if (newRight > furthestRight) {
+				furthestRight = newRight;
+				furthestRightAngle = angleRadians;
+			}
+
+			if (newTop < furthestAbove) {
+				furthestAbove = newTop;
+				furthestAboveAngle = angleRadians;
+			}
+
+			if (newBottom > furthestBelow) {
+				furthestBelow = newBottom;
+				furthestBelowAngle = angleRadians;
+			}
+		}
+
+		var valueCount = getValueCount(scale);
+		for (i = 0; i < valueCount; i++) {
+			pointPosition = scale.getPointPosition(i, largestPossibleRadius);
+			textSize = measureLabelSize(scale.ctx, plFont.size, scale.pointLabels[i] || '');
+			scale._pointLabelSizes[i] = textSize;
+
+			// Add quarter circle to make degree 0 mean top of circle
+			var angleRadians = scale.getIndexAngle(i);
+			var angle = helpers.toDegrees(angleRadians) % 360;
+			determineXLimits(angle);
```

Comment:
Those functions came to make eslint happy otherwise it complains that `fitWithPointLabels` is too complex and has too many statements